src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -319,14 +319,34 @@ public static bool IsKeyValuePair(this Type type, Type? keyValuePairType = null)
 
         public static object? GetDefaultValue(this ParameterInfo parameterInfo)
         {
+            Type parameterType = parameterInfo.ParameterType;
             object? defaultValue = parameterInfo.DefaultValue;
 
+            if (defaultValue is null)
+            {
+                return null;
+            }
+
             // DBNull.Value is sometimes used as the default value (returned by reflection) of nullable params in place of null.
-            if (defaultValue == DBNull.Value && parameterInfo.ParameterType != typeof(DBNull))
+            if (defaultValue == DBNull.Value && parameterType != typeof(DBNull))
             {
                 return null;
             }
 
+#if !BUILDING_SOURCE_GENERATOR
+            // Default values of enums or nullable enums are represented using the underlying type and need to be cast explicitly
+            // cf. https://github.com/dotnet/runtime/issues/68647
+            if (parameterType.IsEnum)
```

Comment:
this is ugly 😄 is there an issue against Type.DefaultValue about it? Why is that necessary?

Reply:
Yep. It's unfortunate but necessary because the method is being used against the source generator's System.Reflection wrappers around Roslyn symbols. These checks are simply not applicable in that world.